File: apps/backend/app/schemas/category.py
from typing import Optional, List
from pydantic import BaseModel, ConfigDict, field_validator
from uuid import UUID

class CategoryBase(BaseModel):
    name: str
    image_url: str | None = None
    commission_percent: Optional[float] = None  # 0..100

    @field_validator("commission_percent")
    @classmethod
    def _check_commission(cls, v):
        if v is None:
            return v
        if v < 0 or v > 100:
            raise ValueError("commission_percent deve estar entre 0 e 100")
        return v

class CategoryCreate(CategoryBase):
    pass

class CategoryRead(CategoryBase):
    id: UUID
    model_config = ConfigDict(from_attributes=True)


class CategoryPage(BaseModel):
    items: List[CategoryRead]
    total: int
    page: int
    size: int

File: apps/clubi.ly/src/types/category.ts
// src/types/category.ts

export interface CategoryBase {
  name: string;
  image_url?: string | null;
  commission_percent?: number | null;
}



/** Categoria retornada pela API */
export interface CategoryRead extends CategoryBase {
  id: string;
}


/** Página pública de categorias */
export interface CategoryPage {
  items: CategoryRead[];
  total: number;
  page: number;
  size: number;
}
